src/core/UserController.ts

```diff
@@ -0,0 +1,47 @@
+import { User } from '../model/User';
+import { UserCreate } from '../model/UserCreate';
+import { UserUpdate } from '../model/UserUpdate';
+import { UserDataAccess } from './UserDataAccess';
+
+export class UserController {
```

Comment:
> What value does this Controller abstraction provide over allowing Lambdas direct access to Data Access layer?

@afaq-sf Yeah, it really isn't doing anything at this point. It is however providing a facade to index.ts which is abstracting the AWS (APIGateway/Lambda) implementations from the rest of the application. If you wanted to pull this application out of lambda tomorrow and deploy it on ECS you'd only need to replace index.ts with a server (express, restify, fastify, nestjs) routing layer.  You could leave `UserController` in place.  This may seem like a thin argument as you could claim the `UserDataAccess` is doing that as well.  But as this application grows the controller is likely to gain some business logic or at least delegate to more downstream dependencies.  In that case it will be easier to replace the lambda layer and test this layer on it's own.  

Perhaps premature optimization here but experience tells me this will be needed soon.